sources/sns/endpoint.go

```diff
@@ -0,0 +1,95 @@
+package sns
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+type endpointClient interface {
+	ListEndpointsByPlatformApplication(ctx context.Context, params *sns.ListEndpointsByPlatformApplicationInput, optFns ...func(*sns.Options)) (*sns.ListEndpointsByPlatformApplicationOutput, error)
+	GetEndpointAttributes(ctx context.Context, params *sns.GetEndpointAttributesInput, optFns ...func(*sns.Options)) (*sns.GetEndpointAttributesOutput, error)
+	ListTagsForResource(context.Context, *sns.ListTagsForResourceInput, ...func(*sns.Options)) (*sns.ListTagsForResourceOutput, error)
+}
+
+func getEndpointFunc(ctx context.Context, client endpointClient, scope string, input *sns.GetEndpointAttributesInput) (*sdp.Item, error) {
+	output, err := client.GetEndpointAttributes(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	if output.Attributes == nil {
+		return nil, &sdp.QueryError{
+			ErrorType:   sdp.QueryError_NOTFOUND,
+			ErrorString: "get endpoint attributes response was nil",
+		}
+	}
+
+	attributes, err := sources.ToAttributesCase(output.Attributes)
+	if err != nil {
+		return nil, err
+	}
+
+	err = attributes.Set("endpointArn", *input.EndpointArn)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceTags, err := tagsByResourceARN(ctx, client, *input.EndpointArn)
```

Comment:
Let's not fail if we can't get tags. Instead if there is an error let's just not set them

Reply:
https://github.com/overmindtech/aws-source/pull/421/commits/51b21c11fb0fee20c9a1318a922f9487b7cd0a7b